### ISSUE TITLE
Improve Dropdown Responsiveness and Interaction in DBeaver

### ISSUE THREAD

**Issue by arsalanhub** [dbeaver/dbeaver#35422]
**Is your feature request related to a problem? Please describe.**
I'm always frustrated when trying to interact with dropdowns in DBeaver. The dropdowns for each table are very slow to respond and only open when the dropdown area turns blue, requiring a very precise click. This inconsistent and slow response time significantly hampers my workflow and efficiency.

**Describe the solution you'd like**
I would like to see an improvement in the responsiveness and usability of dropdown menus in DBeaver. Specifically, the dropdowns should open more quickly and reliably with less precise clicking. The interaction should be smoother and more intuitive, possibly by increasing the clickable area or implementing a more responsive design.

**Describe alternatives you've considered**
I have considered manually entering table names or using keyboard shortcuts to bypass the dropdowns, but these methods are less efficient and do not fully address the root cause of the problem. Additionally, I have explored other database management tools, but I prefer DBeaver for its features and flexibility.

**Additional context**
A smoother and more responsive dropdown menu would greatly enhance the user experience in DBeaver. Improving this feature would save time and reduce frustration during database management tasks.

**Comment by Destrolaric** [comment]
@arsalanhub Hi, could you specify your OS? DBeaver relies on some native os components, and dropdown responsiveness can be different between OSes.

**Comment by arsalanhub** [comment]
@Destrolaric I am using Windows and connecting with SQL Server

**Comment by ShadelessFox** [comment]
@arsalanhub can you please point to specific examples of drop-down menus you have trouble with?

**Comment by arsalanhub** [comment]
@ShadelessFox Suppose if I want to expand Databases toggle then I will need to precisely click on the arrow. Instead it should show me cursor pointer over the toggle for smooth opening of the toggle.

![image](https://github.com/user-attachments/assets/faea76f5-f522-42e6-b3dc-74fae1b3875a)

**Comment by ShadelessFox** [comment]
Well, it's the default tree widget. You can use arrows to navigate and expand tree nodes by pressing **Right**.

> show me cursor pointer over the toggle for smooth opening of the toggle

I'm sorry, but I don't quite understand what you mean by that. What toggle are you referring to? Do you have examples from other apps you can use as a reference?

**Comment by arsalanhub** [comment]
@ShadelessFox I meant that instead of precisely clicking on the arrows can we add the functionality that if the click is made anywhere on that row that contains the arrow? The hidden part should expand.

**Comment by ask9** [comment]
> @ShadelessFox I meant that instead of precisely clicking on the arrows can we add the functionality that if the click is made anywhere on that row that contains the arrow? The hidden part should expand.

I think it is not possible because "on click"-handler is currenlty used for selecting an item in a tree. For example, you can select a few connections, then make use of group functions in context menu for all of them: Copy, Edit, Delete...

**Comment by ask9** [comment]
To enhance user experience, I would suggest to enlarge the clicking area responsible for expand/collapse the items in a tree to the empty space left of the triangle, if it is possible.
I think it would fix your issue and would be easier for other users to click the larger area, not only the tiny triangle.

@arsalanhub
Isn't it?

Make these empty spaces expand/collapse the items in a tree:

![изображение](https://github.com/user-attachments/assets/245a9620-cbb6-4258-a50f-c629bbca05ed)

**Comment by ShadelessFox** [comment]
@arsalanhub you can change the default behavior of double-clicking on a node to expand it:

![image](https://github.com/user-attachments/assets/c5ff3a58-f19e-47ed-b58f-074e30815ccc)

**Comment by arsalanhub** [comment]
> To enhance user experience, I would suggest to enlarge the clicking area responsible for expand/collapse the items in a tree to the empty space left of the triangle, if it is possible.

Yeah, this will be great. @ShadelessFox What do you think?

**Comment by ivgag** [comment]
> To enhance user experience, I would suggest to enlarge the clicking area responsible for expand/collapse the items in a tree to the empty space left of the triangle, if it is possible.

The navigator tree is system control and doesn't support it
